src/core/lv2/Lv2Ports.cpp

```diff
@@ -0,0 +1,259 @@
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include "Lv2Ports.h"
+
+#ifdef LMMS_HAVE_LV2
+
+#include "Engine.h"
+#include "Lv2Basics.h"
+#include "Lv2Manager.h"
+
+namespace Lv2Ports {
+
+
+
+
+const char *toStr(Flow pf)
+{
+	switch(pf)
+	{
+		case Flow::Unknown: return "unknown";
+		case Flow::Input: return "input";
+		case Flow::Output: return "output";
+	}
+	return "";
+}
+
+
+
+
+const char *toStr(Type pt)
+{
+	switch(pt)
+	{
+		case Type::Unknown: return "unknown";
+		case Type::Control: return "control";
+		case Type::Audio: return "audio";
+		case Type::Event: return "event";
+		case Type::Cv: return "cv";
+	}
+	return "";
+}
+
+
+
+
+const char *toStr(Vis pv)
+{
+	switch(pv)
+	{
+		case Vis::Toggled: return "toggled";
+		case Vis::Enumeration: return "enumeration";
+		case Vis::Integer: return "integer";
+		case Vis::None: return "none";
+	}
+	return "";
+}
+
+
+
+
+std::vector<PluginIssue> Meta::get(const LilvPlugin *plugin,
+										unsigned int portNum)
+{
+	std::vector<PluginIssue> portIssues;
+	auto issue = [&portIssues](PluginIssueType i, const char* msg = "") {
+		portIssues.emplace_back(i, msg); };
+
+	Lv2Manager* man = Engine::getLv2Manager();
+	AutoLilvNode connectionOptional = man->uri(LV2_CORE__connectionOptional);
+
+	const LilvPort* lilvPort = lilv_plugin_get_port_by_index(plugin, portNum);
+
+	auto portFunc = [&plugin, &lilvPort, &man](
+		bool (*fptr)(const LilvPlugin*, const LilvPort*, const LilvNode*),
+		const char* str) {
+		bool res = fptr(plugin, lilvPort, man->uri(str).get());
+		return res;
+	};
+
+	auto hasProperty = [&portFunc](const char* str) {
+		return portFunc(lilv_port_has_property, str); };
+	auto isA = [&portFunc](const char* str) {
+		return portFunc(lilv_port_is_a, str); };
+
+
+	const char* portName = lilv_node_as_string(lilv_port_get_name(plugin, lilvPort));
+
+	m_optional = lilv_port_has_property(plugin, lilvPort,
+										connectionOptional.get());
+
+	m_vis = hasProperty(LV2_CORE__integer)
+		? Vis::Integer // WARNING: this may still be changed below
+		: hasProperty(LV2_CORE__enumeration)
+		? Vis::Enumeration
+		: hasProperty(LV2_CORE__toggled)
+		? Vis::Toggled
+		: Vis::None;
+
+	if (isA(LV2_CORE__InputPort)) {
```

Comment:
Explicit blocking according to convention, also for the followng `else if` and `else`